hpcrawler.py

```diff
@@ -0,0 +1,364 @@
+import gpxpy.gpx
+import numpy as np
+import selenium.common
+from selenium import webdriver
+import json
+import os
+import gpxpy
+import pandas as pd
+import SlopeMap as sm
+import re
+
+# TODO: if we want to crawl more data from website after we've already established a data set,
+#       we need to :
+#       (1) fully re-implement the functions:
+#               * get_page_data -> EASY
+#               * process_track_data -> EASY
+#           to supply the desired functionality.
+#       (2) partially re-implement the function:
+#               * collect_track_data -> LESS EASY
+#           we'll copy the <progress> file, we'll call it <p>.
+#           while questioning the web about the new features we'll get the track's name and
+#           compare it to the ones in <p>, and we will update the corresponding track in <p>
+#           to include the new features. if a track isn't on <p> , we'll mine it all up.
+#       (3) create different:
+#               * directory <tracks_dir_path>: we don't need to overwrite the whole data set.
+#               * file <p>
+#           so maybe we should give them as parameters to the class?
+#
+
+
+class HpCrawler:
+    """
+    This class crawls "The Hiking Project"'s website: https://www.hikingproject.com/.
+    It process the data crawled, and saves it, as follows:
+    it creates a dir named "hp", that contains 2 dirs and a file:
+     (1) dir <gpx_dir_path> : contains country directories, each directory holds all gpx files of the tracks
+                            crawled from that country and a json file keeping trace of the data collecting progress
+                            * The dir is named by the country name, string: <country>
+                            * The gpx files are named by the track's idx, int >= 0: <j>
+                            * The json file is named <progress>: maps [track_gpx_filename, track_length,
+                                track_difficulty] to track idx <j>, int >=0.
+     (2) dir <tracks_dir_path> : contains json files. These files are named by the length_tag of the tracks
```

Comment:
variable name is unclear. would suggest something like sorted_track_dir (also dir_path is odd since they mean almost the same thing)